src/main/java/org/jenkins_ci/plugins/run_condition/core/StringsMatchCondition.java

```diff
@@ -59,11 +68,25 @@ public boolean isIgnoreCase() {
 
     @Override
     public boolean runPerform(final AbstractBuild<?, ?> build, final BuildListener listener) throws Exception {
-        final String expanded1 = TokenMacro.expandAll(build, listener, arg1);
-        final String expanded2 = TokenMacro.expandAll(build, listener, arg2);
-        listener.getLogger().println(Messages.stringsMatchCondition_console_args(expanded1, expanded2));
-        if (expanded1 == null) return false;
-        return ignoreCase ? expanded1.equalsIgnoreCase(expanded2) : expanded1.equals(expanded2);
+
+        if(isshellVariables() == true){
+
+             final String expanded1 = TokenMacro.expandAll(build, listener, "$"+"{"+arg1+"}");
+             final String expanded2 = TokenMacro.expandAll(build, listener, "$"+"{"+arg2+"}");
```

Comment:
This seems like it only handles the Unix environment variables expansion.  Aren't Windows environment variables represented as well?  Something like `%MY_VAR%`?

Reply:
Yes, Sir they are handled